### PR TITLE
Booth/dev

### PR DESCRIPTION
**Pull request by jeremiahbooth-insight**
Made the inputs to be more modular and added them to the completed forms. Form Three is not complete yet will have another pull request with that one is ready

### CODE REVIEW

File: UI/libs/shared/ui/src/lib/validators/form-validators/formStepOneValidators.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+type CheckerObject = {
+  personalInfo: PersonalInfo;
+  ssnConfirm: string;
+};
+
+export function validateFormStepOne(formData: CheckerObject) {
+  const errors: Array<string> = [];
+  if (!formData.personalInfo.lastName) {
+    errors.push('Last name');
```
Comment (sgaddameedi-insight) [inline UI/libs/shared/ui/src/lib/validators/form-validators/formStepOneValidators.ts:11]: @jeremiahbooth-insight  We should get rid of the validators folder as well, it's very hard to navigate/read and maintain for a large number of forms and inputs - In order to make components reusable. Follow the below approach:

```vue
<template>
  <v-form>
    <v-container>
      <v-row>
        <v-col
          cols="12"
          sm="6"
        >
          <v-text-field
            v-model="email"
            :rules="[rules.required, rules.email]"
            label="E-mail"
          ></v-text-field>
        </v-col>
      </v-row>
    </v-container>
  </v-form>
</template>
```

```ts
<script>
  export default {
    data () {
      return {
        title: 'Email',
        email: '',
        rules: {
          required: value => !!value || 'Required.',
          counter: value => value.length <= 20 || 'Max 20 characters,
          email: value => {
            const pattern = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
            return pattern.test(value) || 'Invalid e-mail.'
          },
        },
      }
    },
  }
</script>
```

This way, we won't be limited.

File: UI/libs/shared/ui/src/lib/validators/form-validators/formStepTwoValidators.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import { Citizenship, DOB, Id } from '@shared-ui/types/defualtTypes';
+type checkerObject = {
+  id: Id;
+  dob: DOB;
+  citizenship: Citizenship;
+};
+
+export function validateForm(formData: checkerObject) {
```
Comment (sgaddameedi-insight) [inline UI/libs/shared/ui/src/lib/validators/form-validators/formStepTwoValidators.ts:8]: Same here

**Review by sgaddameedi-insight — CHANGES_REQUESTED** [review]

**Review by sgaddameedi-insight — CHANGES_REQUESTED** [review]
Also - Can you please rebase with main? You will most likely see few conflicts and have to fixup on top of it. Thanks

File: UI/libs/shared/ui/src/lib/helpers/storeHelpers/citizenActions.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { store } from '@shared-ui/store';
+import { Citizenship } from '@shared-ui/types/defualtTypes';
+
+export async function addCitizenshipAction(citizenship: Citizenship) {
+  await store.dispatch('addCitizenshipInfo', citizenship);
```
[the remark this answers is not in the document]
Comment (jeremiahbooth-insight) [reply UI/libs/shared/ui/src/lib/helpers/storeHelpers/citizenActions.ts:5]: They are I just forgot to delete the action files one sec

[the remark this answers is not in the document]
Comment (sgaddameedi-insight) [reply UI/libs/shared/ui/src/lib/helpers/storeHelpers/citizenActions.ts]: Thanks

File: UI/libs/shared/ui/src/lib/components/tables/AddressTable.vue [hunk 4]
```diff
@@ -0,0 +1,60 @@
+<template>
+  <v-simple-table class="my-3">
+    <template #default>
+      <thead>
+        <tr>
+          <th>Address line 1</th>
```
Comment (sgaddameedi-insight) [inline UI/libs/shared/ui/src/lib/components/tables/AddressTable.vue:6]: Add $t translation format here

File: UI/libs/shared/ui/src/lib/components/form-stepper/form-steps/FormStepTwo.vue [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+        :disabled="!valid"
+        color="secondary mr-2"
+        @click="handleSubmit"
+      >
+        Continue
+      </v-btn>
+      <!-- TODO: this needs to save to save the current state to local storage or call the api to save in
+      the db
+       -->
+      <v-btn color="info mr-2">Save and Exit</v-btn>
```
Comment (sgaddameedi-insight) [inline UI/libs/shared/ui/src/lib/components/form-stepper/form-steps/FormStepTwo.vue:154]: Add $t translation format here

File: UI/libs/shared/ui/src/lib/components/form-stepper/form-steps/FormStepTwo.vue [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+            v-model="DOBInfo.birthCountry"
+            label="Birth country"
+            required
+            :rules="[v => !!v || 'Birth country cannot be blank']"
+          />
+        </v-col>
+      </v-row>
+
+      <v-divider />
+      <v-subheader class="subHeader font-weight-bold"> Citizenship Information </v-subheader>
```
Comment (sgaddameedi-insight) [inline UI/libs/shared/ui/src/lib/components/form-stepper/form-steps/FormStepTwo.vue:85]: Add $t translation format here

File: UI/libs/shared/ui/src/lib/components/form-stepper/form-steps/FormStepTwo.vue [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+            v-model="id.issuingState"
+            label=" Issuing State"
+            :rules="[v => !!v || 'Issuing state is required']"
+            required
+          />
+        </v-col>
+      </v-row>
+
+      <v-divider />
+      <v-subheader class="subHeader font-weight-bold"> Date of birth </v-subheader>
```
Comment (sgaddameedi-insight) [inline UI/libs/shared/ui/src/lib/components/form-stepper/form-steps/FormStepTwo.vue:37]: @jeremiahbooth-insight  - can u add $t to all static strings? I think eslint should direct you the locations where it's missing. try

```bash
nx affected --target=lint --all --fix --skip-nx-cache 
```

File: UI/libs/shared/ui/src/lib/validators/form-validators/formStepOneValidators.ts (hunk 1 above)
Comment (sgaddameedi-insight) [reply UI/libs/shared/ui/src/lib/validators/form-validators/formStepOneValidators.ts]: @jeremiahbooth-insight  Looks like this is resolved. PR looks much better, Thanks!

**Review by sgaddameedi-insight — CHANGES_REQUESTED** [review]

**Review by sgaddameedi-insight — APPROVED** [review]
LGTM - Appreciate your effort

**Pull request merged**